python_modules/dagster/dagster/_core/execution/submit_asset_runs.py

```diff
@@ -142,47 +148,63 @@ def _create_asset_run(
                 workspace=workspace,
                 run_request_execution_data_cache=run_request_execution_data_cache,
             )
-            check_for_debug_crash(debug_crash_flags, "EXECUTION_PLAN_CREATED")
-            check_for_debug_crash(debug_crash_flags, f"EXECUTION_PLAN_CREATED_{run_request_index}")
 
-            # retry until the execution plan targets the asset selection
-            if _execution_plan_targets_asset_selection(
-                execution_data.external_execution_plan.execution_plan_snapshot,
-                check.not_none(run_request.asset_selection),
+        except (DagsterInvalidSubsetError, DagsterUserCodeProcessError):
+            logger.warning(
+                "Error while generating the execution plan, possibly because the code server is out of sync with the daemon. Pausing run submission for "
+                f"{RELOAD_WORKSPACE_INTERVAL} seconds to allow the execution plan to rebuild with the updated workspace.",
+                exc_info=sys.exc_info(),
+            )
```

Comment:
Can you add:

"The daemon periodically refreshes its representation of the workspace and the pause allows time for that to happen" or something to that effect?

"to allow the execution plan to rebuild with the updated workspace." sounds extremely magical and is confusing